torchvision/transforms/functional.py

```diff
@@ -945,26 +945,42 @@ def adjust_gamma(img: Tensor, gamma: float, gain: float = 1) -> Tensor:
 
 
 def _get_inverse_affine_matrix(
-    center: List[float], angle: float, translate: List[float], scale: float, shear: List[float]
+    center: List[float],
+    angle: float,
+    translate: List[float],
+    scale: float,
+    shear: List[float],
+    centered_shear: bool = True,
 ) -> List[float]:
     # Helper method to compute inverse matrix for affine transformation
 
-    # As it is explained in PIL.Image.rotate
-    # We need compute INVERSE of affine transformation matrix: M = T * C * RSS * C^-1
+    # Pillow requires inverse affine transformation matrix:
+    #  option 1 (centered_shear=True) curr  : M = T * C * RotateScaleShear * C^-1
+    #  option 2 (centered_shear=False) new  : M = T * C * RotateScale * C^-1 * Shear
+    #
     # where T is translation matrix: [1, 0, tx | 0, 1, ty | 0, 0, 1]
     #       C is translation matrix to keep center: [1, 0, cx | 0, 1, cy | 0, 0, 1]
-    #       RSS is rotation with scale and shear matrix
-    #       RSS(a, s, (sx, sy)) =
+    #       RotateScaleShear is rotation with scale and shear matrix
+    #       RotateScale is rotation with scale matrix
+    #
+    #       RotateScaleShear(a, s, (sx, sy)) =
     #       = R(a) * S(s) * SHy(sy) * SHx(sx)
-    #       = [ s*cos(a - sy)/cos(sy), s*(-cos(a - sy)*tan(x)/cos(y) - sin(a)), 0 ]
-    #         [ s*sin(a + sy)/cos(sy), s*(-sin(a - sy)*tan(x)/cos(y) + cos(a)), 0 ]
+    #       = [ s*cos(a - sy)/cos(sy), s*(-cos(a - sy)*tan(sx)/cos(sy) - sin(a)), 0 ]
+    #         [ s*sin(a + sy)/cos(sy), s*(-sin(a - sy)*tan(sx)/cos(sy) + cos(a)), 0 ]
     #         [ 0                    , 0                                      , 1 ]
     #
+    #       RotateScale(a, s) =
+    #       = R(a) * S(s)
+    #       = [ s*cos(a), -s*sin(a), 0 ]
+    #         [ s*sin(a),  s*cos(a), 0 ]
+    #         [ 0       ,  0       , 1 ]
+    #
     # where R is a rotation matrix, S is a scaling matrix, and SHx and SHy are the shears:
     # SHx(s) = [1, -tan(s)] and SHy(s) = [1      , 0]
     #          [0, 1      ]              [-tan(s), 1]
-    #
-    # Thus, the inverse is M^-1 = C * RSS^-1 * C^-1 * T^-1
+
+    # TODO: implement the option
+    assert centered_shear
```

Comment:
I would recommend to remove this and open an issue where we provide the details of why we should implement this option, why it's necessary and references to papers or other implementations that follow this approach.